src/vm/runners/builtin_runner/ec_op.rs

```diff
@@ -1032,38 +1037,50 @@ mod tests {
     }
 
     #[test]
-    fn catch_point_not_in_curve() {
-        let program = Program::from_file(
-            Path::new("cairo_programs/bad_programs/ec_op_not_in_curve.json"),
-            Some("main"),
-        )
-        .expect("Call to `Program::from_file()` failed.");
-
-        let mut hint_processor = BuiltinHintProcessor::new_empty();
-        let mut cairo_runner = cairo_runner!(program, "all", false);
-        let mut vm = vm!();
-
-        let end = cairo_runner.initialize(&mut vm).unwrap();
-        assert!(cairo_runner
-            .run_until_pc(end, &mut vm, &mut hint_processor)
-            .is_err());
+    fn catch_point_same_x() {
+        let program = Path::new("cairo_programs/bad_programs/ec_op_same_x.json");
+        let result = crate::cairo_run::cairo_run(
+            program,
+            "main",
+            false,
+            false,
+            "all",
+            false,
+            None,
+            &mut BuiltinHintProcessor::new_empty(),
+        );
+        assert!(result.is_err());
+        // We need to check this way because CairoRunError doenst implement Debug
```

Comment:
Does anything stop us from deriving `Debug`?

Reply:
Typo:
```suggestion
        // We need to check this way because CairoRunError doesn't implement Debug
```

Reply:
Good catch!

Reply:
I think its because ProgramError uses serde's errors which dont implement Debug, we could probably implement some mapping for them (in another PR)

Reply:
This is true for PartialEq, not Debug, Ill fix the comment